src/gnutls_session.cpp

```diff
@@ -0,0 +1,350 @@
+#include "connection.hpp"
+#include "gnutls_crypto.hpp"
+
+namespace oxen::quic
+{
+    extern "C"
+    {
+        int gnutls_callback_wrapper(
+                gnutls_session_t session,
+                unsigned int htype,
+                unsigned int when,
+                unsigned int incoming,
+                const gnutls_datum_t* msg)
+        {
+            auto* tls_session = get_session_from_gnutls(session);
+            // No need to assert the ptr is set here, it's checked in the function before returning.
+            // Instead, check that the return matches expected
+            assert(tls_session->get_session() == session);
+
+            return tls_session->do_tls_callback(session, htype, when, incoming, msg);
+        }
+
+        int gnutls_post_handshake(gnutls_session_t session)
+        {
+            auto* tls_session = get_session_from_gnutls(session);
+            // Same notes on the assert as in gnutls_callback_wrapper above
+            assert(tls_session->get_session() == session);
+            (void)tls_session;
```

Comment:
An alternative to these three lines that won't hit the unused variable warning is to inline it all inside the assert (sometimes that gets hairy, but here seems fairly innocuous):
```c++
assert(get_session_from_gnutls(session)->get_session() == session);
```

Reply:
this actually still flipped an unused variable error for the parameter session